app/helpers/runnables_helper.rb

```diff
@@ -45,16 +45,16 @@ def student_run_buttons(offering,opts={})
       end
       if display_workgroups_run_link?(offering)
         options[:class] = student_run_button_css(offering, ["in_group"])
-        # These attributes use by AngularJS code, see: angular/collaboration.js.coffee
-        options[:'cc-setup-collaboration'] = true
-        options[:'data-offering-id'] = offering.id
         # Collaboration setup works differently for external activities and JNLP ones.
         # jnlp-url attribute lets us distinguish between them.
         if !offering.external_activity?
           options[:'data-jnlp-url'] = run_url_for(offering)
         end
-        haml_tag :a, options do
-          haml_concat group_label
+        options[:label] = group_label
+        options[:offeringId] = offering.id
+        haml_tag :span, {:id => 'run-with-collaborators-button'}
+        haml_tag :script do
+          haml_concat "PortalPages.renderRunWithCollaborators(#{options.to_json}, 'run-with-collaborators-button');"
```

Comment:
Won't this fail if there are multiple buttons on the same page, since it is looking up the span by a fixed id?